tests/operator/suite.go

```diff
@@ -126,6 +126,53 @@ func LoadChecks() {
 			testOperatorCatalogSourceBundleCount(c, &env)
 			return nil
 		}))
+
+	checksGroup.Add(checksdb.NewCheck(identifiers.GetTestIDAndLabels(identifiers.TestOperatorInstallationInTenantNamespace)).
+		WithSkipCheckFn(testhelper.GetNoOperatorsSkipFn(&env)).
+		WithCheckFn(func(c *checksdb.Check) error {
+			testOperatorInstallationInTenantNamespace(c, &env)
+			return nil
+		}))
+}
+
+// This function checks if SingleNamespaced Operators should only be installed in the tenant dedicated operator namespace
+func testOperatorInstallationInTenantNamespace(check *checksdb.Check, env *provider.TestEnvironment) {
+	check.LogInfo("Starting testInstalledSingleNamespaceOperatorInTenanttNamespace")
+
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
+
+	singleNamespacedOperators := filterSingleNamespacedOperatorUnderTest(env.Operators)
+
+	// namespaces to be considered
+	var namespaces []string
+	for i := range singleNamespacedOperators {
+		operator := singleNamespacedOperators[i]
+		namespaces = append(namespaces, operator.Namespace)
+	}
+
+	for _, operatorNamespace := range namespaces {
+		check.LogInfo("Checking if namespace %s is an operator namespace", operatorNamespace)
+		doesNamespaceContainsOnlyOperatorPods, csvs, failedPods, err := checkIfNamespaceContainsOnlyOperatorPods(operatorNamespace)
+
+		if err != nil {
+			check.LogError("Skipped - cannot proceed with operator namespace %s", operatorNamespace)
+			continue
+		}
+
+		if doesNamespaceContainsOnlyOperatorPods {
+			for _, csv := range csvs {
+				check.LogInfo("Csv %s has valid installation in tenant namespace %s ", csv, operatorNamespace)
+				compliantObjects = append(compliantObjects, testhelper.NewOperatorReportObject(operatorNamespace, csv,
+					"Operator has valid installation in tenant namespace ", true))
+			}
+		} else {
+			check.LogInfo("Operator namespace %s contains mixed invalid operator pods ", operatorNamespace)
+			nonCompliantObjects = append(nonCompliantObjects, testhelper.NewReportObject(operatorNamespace,
```

Comment:
Why not using `testhelper.NewNamespacedReportObject` instead?

Reply:
Updated!